compiler/AST/build.cpp

```diff
@@ -378,11 +379,79 @@ static BlockStmt* buildUseList(BaseAST* module, BlockStmt* list) {
   }
 }
 
+//
+// Given an argument from a 'use' statement, process that argument
+// if it's a string (returning whether or not it was).  
+//
+// If it's a string literal, we assume it's either a "-llib" flag,
+// or a filename like "foo.h", "foo.c", "foo.o", etc.  
+//   - Peel off the former and pass them to addLibInfo(), the same
+//     function that handles command line -l flags.
+//   - Otherwise, assume it's the latter and pass it to our input
+//     file handler (which itself handles cases it doesn't recognize.
+//
+static bool processUseOfString(BaseAST* arg) {
+  //
+  // If we're 'use'ing a literal string...
+  //
+  if (SymExpr* se = toSymExpr(arg)) {
+    if (VarSymbol* var = toVarSymbol(se->var)) {
+      if (var->isImmediate()) {
+        Immediate* imm = var->immediate;
+        if (imm->const_kind == CONST_KIND_STRING) {
+          //
+          // And it starts with -l, treat it as a library specifier
+          //
+          if (strncmp(imm->v_string, "-l", 2) == 0) {
+            addLibInfo(astr(imm->v_string));
```

Comment:
I don't think you need the `astr()` here. `addLibInfo()` does this for you:  https://github.com/bradcray/chapel/blob/use-other-stuff/compiler/util/files.cpp#L75